sureswiftcapital/Simple-Calendar
# 13498288 google referer restrictions

This fixes an issue where restricting Google Calendar API referers when creating an API key would cause calendar display to break. This fix adds `referer` headers to the Google Client [here](https://github.com/sureswiftcapital/Simple-Calendar/compare/master...13498288-google-referer-restrictions?expand=1#diff-95bb045948343939383803deac3d8218R142).

Ref: [Trello](https://trello.com/c/BMRh7LQt) / [Teamwork](https://codivated.teamwork.com/#tasks/13498288)

This also updates `npm` dependencies as well as bumps the composer Google Client library from v1.x to v2.x.

These changes should be thoroughly tested for compatibility, including with all internal add-ons (Fullcalendar, Simple Calendar Pro, etc.). Most breaking client library changes pertaining to v1 -> v2 upgrade relate to items beyond just reading the Google Calendar feed data, so impact here shouldn't be an issue. I tested myself, but would like another set of eyes. More info [here](https://github.com/googleapis/google-api-php-client/blob/master/UPGRADING.md).

**Testing scenarios:**

Since this fix if for issues with API referer restrictions, in order to test, an API key must first be restricted via Google Console [here](https://console.developers.google.com/apis/credentials?project={your_project}). For example if testing on https://codivated.com/calendar for example, restrict specifically to that url or use wildcards such as in this screenshot...

<img width="1202" alt="Screen Shot 2019-03-19 at 9 24 01 PM" src="https://user-images.githubusercontent.com/867556/54652574-b8a25000-4a8d-11e9-9c3c-2fbe75ffbd42.png">

Keep in mind that after setting the restriction it can take a few minutes before it becomes active. Once that is set and active, test the following using both Google Calendar free as well as with the Pro & FullCalendar addons...

1. Can you create a calendar without issue?
1. Embedding the calendar in a page or post works correctly.
1. Try a variety of settings... this includes grid, list, fullcalendar, pro, free, grouped calendars, etc.
     1. do they all work? If not, what combination(s) do not work, and what errors do you see?
1. Try a variety of events? Adding and removing new events in Google works okay and shows up on the WP end okay (once calendar cache is cleared -- event changes won't show unless this is done).

Reply with a comment below if you have questions.
